### PR TITLE
Improve readability of Core & Extension testing chapters

### PR DESCRIPTION
**Pull request by tomwarwick**
This commit is intended to improve the readability of two chapters within the Testing section of this document. This commit contains changes that aim to:

- Ensure that the opening statement for each chapter details its intent in a clearer and more concise manner
- Create a consistent and more formal tone
- Other minor changes, including capitalising product names

### CODE REVIEW

File: Documentation/Testing/CoreTesting.rst [hunk 1]
```diff
@@ -9,20 +9,17 @@ Core testing
...
 since core development is most likely bound to the core master branch - back porting patches to older
-branches are usually handled by core maintainers and often don't affect other core contributors much.
+branches are usually handled by core maintainers and often don't affect other core contributors.
 
-As a disclaimer, the main script :file:`Build/Scripts/runTests.sh` is relatively young. It works best
-if executed on a linux based host, it works using macOS (but much slower), and it works using
-Windows (and it's actually quicker on Windows than with mac). Not everything in this area is settled yet,
-though.
+Note, the main script :file:`Build/Scripts/runTests.sh` is relatively new. It works best
+when executed on a Linux based host but can be run under macOS and Windows with some performance drawbacks.
```
Comment (sypets) [inline Documentation/Testing/CoreTesting.rst]: Correct would be: ... macOS (with some performance drawbacks) and Windows. 

The performance problems only apply to macOS

File: Documentation/Testing/CoreTesting.rst [hunk 2]
```diff
@@ -69,37 +66,37 @@ system. Executing the basic core unit test suite boils down to:
     Build/Scripts/runTests.sh
 
 That's it. You just executed the entire unit test suite. How long did test execution take on your side?
-Your local environment is cool if you are below thirty seconds ... ok, let's say two minutes on mac. After
-initial core clone and a composer install, other parts of this chapter are about different permutations of
-calling `runTests.sh` only.
+initial core clone and a composer install, other parts of this chapter are about different permutations of.
+Now that we have examined the initial core clone and a composer install process, we will now look at the
```
Comment (sypets) [inline Documentation/Testing/CoreTesting.rst]: - Is something missing here? After "How long did test execution take on your side?"
- remove second "now": "Now that we have examined ..., we will now look at."

File: Documentation/Testing/CoreTesting.rst [hunk 3]
```diff
@@ -69,37 +66,37 @@ system. Executing the basic core unit test suite boils down to:
...
 :file:`Build/bamboo/src/main/java/core/PreMergeSpec.java` and :file:`Build/testing-docker/bamboo/docker-compose.yml`
-specify what bamboo executes for patches pushed to the review system. On local testing, this is the
+specify what Bamboo executes for patches pushed to the review system. On local testing, this is the
 combination of :file:`Build/Script/runTests.sh`, :file:`Build/testing-docker/local/.env` (created by
 runTests.sh) and :file:`Build/testing-docker/local/docker-compose.yml`.
 
-The cool thing is that runTests.sh can do everything locally that bamboo executes as `pre-merge
-<https://bamboo.typo3.com/browse/CORE-GTC>`_ tests, too. It's just that the combinations of tests
-and splitting to different jobs is slightly different, for instance bamboo does multiple tests in
+Whats impressive is that runTests.sh can do everything locally that Bamboo executes as `pre-merge
```
Comment (sypets) [inline Documentation/Testing/CoreTesting.rst]: What's impressive

File: Documentation/Testing/CoreTesting.rst (hunk 1 above)
Comment (tomwarwick) [reply Documentation/Testing/CoreTesting.rst]: Fair comment, I've updated the paragraph.

File: Documentation/Testing/CoreTesting.rst (hunk 2 above)
Comment (tomwarwick) [reply Documentation/Testing/CoreTesting.rst]: Fixed :)

**Pull request merged**